.buildkite/scripts/lib/packer.sh

```diff
@@ -1,15 +1,41 @@
 #!/usr/bin/env bash
+set -euo pipefail
+
+source .buildkite/scripts/lib/packer_constants.sh
 
 # Localizing the core "logic" (such as it is) for building our AMI. We
 # use this both to build locally and in CI.
 build_ami() {
     # Example usage:
-    # PACKER_VARS="-var build_ami=false" build_ami
+    # PACKER_VARS="-var build_ami=false" build_ami grapl-nomad-consul-server
+    # PACKER_VARS="-var build_ami=false" build_ami grapl-nomad-consul-client
+
+    local -r packer_image_name="$1"
+
+    if [ "${packer_image_name}" == "${PACKER_IMAGE_NAME_SERVER}" ]; then
+        # shellcheck disable=SC2086
+        packer build ${PACKER_VARS:-} -var is_server=true packer/nomad-server-client
+    elif [ "${packer_image_name}" == "${PACKER_IMAGE_NAME_CLIENT}" ]; then
+        # shellcheck disable=SC2086
+        packer build ${PACKER_VARS:-} -var is_server=false packer/nomad-server-client
+    else
+        echo "Unknown PACKER_IMAGE_NAME ${packer_image_name}"
+        exit 42
+    fi
+}
```

Comment:
Since there are only two choices (it's either a server or its not), it might be simpler to pull this naming logic into the Packer template itself, rather than having to contend with making sure we've used the right concrete names everywhere else.

Reply:
1. On one hand, I agree
2. On the other hand, I don't want to code us into a corner where we only support "true" or "false"; we could potentially have a situation in the future where we have >2 AMIs. 

Ultimately, I really wanted some HCL way to either:
- define a string enum of <only 2 valid values right now> 
- or throw an error if a string does not equal "value1" or "value2"

buuuut i couldn't find that

Reply:
found it


  validation {
    condition     = length(var.image_id) > 4 && substr(var.image_id, 0, 4) == "ami-"
    error_message = "The image_id value must be a valid AMI id, starting with \"ami-\"."
  }